Content.Server/Chat/Systems/LastMessageBeforeDeathSystem.cs

```diff
@@ -0,0 +1,151 @@
+                // Example: Someone was giving a speech and got blown to bits.
+                var randomLength = _random.Next(1, MaxICLength);
+                message = message[..randomLength] + "-";
+            }
+
+            var currentTime = _gameTicker.RoundDuration();
+            var truncatedTime = $"{currentTime.Hours:D2}:{currentTime.Minutes:D2}:{currentTime.Seconds:D2}";
+            var formattedMessage = $"[{truncatedTime}] {playerName}: {message}";
+
+            if (!_playerData.Contains(player))
+            {
+                _playerData[player] = new OrderedDictionary();
+            }
+
+            var playerMessages = _playerData[player] as OrderedDictionary ?? throw new InvalidOperationException("Player messages dictionary is not initialized.");
+            playerMessages[playerName] = formattedMessage;
+            playerMessages["EntityUid"] = source;
+            playerMessages["NetUserId"] = _userNetId;
+        }
+
+        /// <summary>
+        ///     Sends all collected messages through the webhook at the end of the round.
+        ///     Skips messages from banned users.
+        /// </summary>
+        /// <param name="webhookId">The identifier of the webhook to send messages to.</param>
+        public async void OnRoundEnd(WebhookIdentifier? webhookId)
+        {
+            if (!webhookId.HasValue)
+                return;
+
+            var id = webhookId.Value;
+            StringBuilder allMessagesString = new StringBuilder();
+            foreach (DictionaryEntry player in _playerData)
+            {
+                var playerMessages = player.Value as OrderedDictionary;
+                if (playerMessages != null)
+                {
+                    if (playerMessages["EntityUid"] is EntityUid playerUid && _mobStateSystem.IsDead(playerUid))
+                    {
+                        if (playerMessages["NetUserId"] is NetUserId userNetId)
+                        {
+                            var ban = await _db.GetServerBanAsync(null, userNetId, null);
+                            if (ban != null)
+                            {
+                                continue; // Skip banned users
+                            }
+                        }
+                        foreach (DictionaryEntry playerName in playerMessages)
+                        {
+                            if (playerName.Key.ToString() != "EntityUid" && playerName.Key.ToString() != "NetUserId")
+                            {
+                                allMessagesString.AppendLine($"{playerName.Value}");
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (allMessagesString.ToString().Length == 0)
+            {
+                allMessagesString.AppendLine($"No messages found.");
+            }
+
+            var messages = _webhookManager.SplitMessage(allMessagesString.ToString(), LastMessageWebhookManager.MaxMessageSize);
```

Comment:
You are concatenating everything and then splitting it again. This is a mess.

Reply:
My bad, now I add all messages to a list and then format them.